tasks/sign-binaries/sign-binaries.yaml

```diff
@@ -0,0 +1,150 @@
+
+    # - name: sign-mac-binaries
+    #   image: quay.io/konflux-ci/release-service-utils:e633d51cd41d73e4b3310face21bb980af7a662f
+    #   script: |
+    #     #!/bin/bash
+    #     # TODO CLDX-79
+    - name: sign-windows-binaries
+      image: quay.io/konflux-ci/release-service-utils:e633d51cd41d73e4b3310face21bb980af7a662f
+      volumeMounts:
+        - name: windows-ssh-key-vol
+          mountPath: "/etc/secrets"
+          readOnly: true
+      env:
+        - name: WINDOWS_USER
+          valueFrom:
+            secretKeyRef:
+              name: $(params.windowsCredentials)
+              key: username
+        - name: WINDOWS_PORT
+          valueFrom:
+            secretKeyRef:
+              name: $(params.windowsCredentials)
+              key: port
+        - name: WINDOWS_HOST
+          valueFrom:
+            secretKeyRef:
+              name: $(params.windowsCredentials)
+              key: host
+        - name: QUAY_USER
+          valueFrom:
+            secretKeyRef:
+              name: $(params.quaySecret)
+              key: username
+        - name: QUAY_PASS
+          valueFrom:
+            secretKeyRef:
+              name: $(params.quaySecret)
+              key: password
+      script: |
+        #!/bin/bash
+        mkdir -p ~/.ssh
+        cp "/etc/secrets/id_rsa" ~/.ssh/id_rsa
+        chmod 600 ~/.ssh/id_rsa
+        unsigned_digest=$(cat "$(results.unsignedWindowsDigest.path)")
+        # EOF should be unquoted here to allow for variable expansion on the client side
+        # shellcheck disable=SC2086,SC2087
+        ssh -i ~/.ssh/id_rsa -o StrictHostKeyChecking=no -p "${WINDOWS_PORT}" "${WINDOWS_USER}@${WINDOWS_HOST}" << EOF
+
+        set SIGNTOOL="C:\Program Files (x86)\Windows Kits\10\bin\10.0.20348.0\x64\signtool.exe"
+        set ORAS="C:\Program Files\Common Files\bin\oras.exe"
+
+        mkdir %TEMP%\$(params.pipelineRunUid) && cd /d %TEMP%\$(params.pipelineRunUid)
+        %ORAS% login quay.io -u ${QUAY_USER} -p ${QUAY_PASS}
+        %ORAS% pull $(params.quayURL)/unsigned@${unsigned_digest}
+
+        %SIGNTOOL% sign /v /n "Red Hat" /fd SHA256 /tr http://timestamp.digicert.com /td SHA256 ^
+        %TEMP%\$(params.pipelineRunUid)\*
+
+        if errorlevel 1 (
+          echo Signing of binaries failed
+          exit /B %ERRORLEVEL%
+        )
+
+        %SIGNTOOL% verify /v /pa %TEMP%\$(params.pipelineRunUid)\*
+
+        if errorlevel 1 (
+          echo Verification of binaries failed
+          exit /B %ERRORLEVEL%
+        )
+
+        echo [%DATE% %TIME%] Signing of Windows binaries completed successfully
+
+        %ORAS% push $(params.quayURL)/signed:$(params.pipelineRunUid) . > oras_push_output.txt 2>&1
+        for /f "tokens=2,3 delims=: " %a in ('findstr "Digest:" oras_push_output.txt') do @echo %a:%b > digest.txt
+
+        EOF
+        DIGEST=$(ssh -i ~/.ssh/id_rsa -o StrictHostKeyChecking=no -p "${WINDOWS_PORT}" \
```

Comment:
I wonder if potentially the ssh server won't print some greetings or something at the top. Wouldn't it be better to just scp the file?

Reply:
Pushing other changes up. Keep this comment unresolved and I'll test SCP'ing the file over in the meantime. Thanks!

Reply:
Now SCP'ing the file over. Good suggestion thanks

Reply:
I see scp is used to copy over the signing script to the Windows host. But this was about copying the digest back and I still see this being done via ssh + type.

Reply:
Ah, yea might as well use SCP that direction too. Thanks done.

Reply:
(Sorry for the delay, I was out.)

Thanks. But this brings another question: Where does the binary come from originally? That part is not done yet, but I assume the binary will first be fetched from somewhere and then the task will push it using oras. But now that we know we can use scp, wouldn't it make more sense to simply copy the binary using scp? Isn't using oras pull/push adding an unnecessary overhead? Or are there any benefits that I'm missing?

Reply:
Fair question.  This was covered in the [design doc](https://docs.google.com/document/d/1FmJmOinT6BSkJiK_AWK3PFDBF_08UqSQUakkivKhKAk/edit?usp=sharing) and approved by Ralph, Brian and others.  

We considered SCP, NFS share and other methods for sharing content. 

If you have questions that aren't covered in the doc or the linked Slack thread from the doc can you ping me on Slack? That might be more efficient and I can summarize our conversation in a comment here afterwards for transparency.

Reply:
Thanks for the link. I sent you a PM with more details, but alright, let's go with oras pull/push then.